### PR TITLE
filter by system

### PR DESCRIPTION
**Pull request by Loui4**

### CODE REVIEW

File: common/models/facility.js [hunk 1]
```diff
@@ -729,4 +729,40 @@ module.exports = Facility => {
...
+        })
+      })
+
+    } catch (error) {
+      cb(error)
+
+    }
+
+
+    return facility
```
Comment (bwighane) [inline common/models/facility.js]: The return type should be a facility and not an array of facilities. Looks like the resolved implementation shoots back an array.

Comment (bwighane) [inline common/models/facility.js]: The payload being returned needs to be properly serialized. The facility_code_mappings entry has the JSON in text format. `"[{\"url\": \"https://dhis2.health.gov.mw/api/organisationUnits/zYfP0gkHRJH.json\", \"code\": \"zYfP0gkHRJH\", \"system\": \"DHIS2\"}, {\"url\": \"https://lmis.health.gov.mw\", \"code\": \"LL4040\", \"system\": \"OpenLMIS\"}, {\"url\": \"https://lmis.health.gov.mw\", \"code\": \"LL4040\", \"system\": \"OpenLMIS\"}, {\"url\": \"https://www.hiv.health.gov.mw/\", \"code\": \"702\", \"system\": \"DHAMIS\"}]"`

File: common/models/facility.js [hunk 2]
```diff
@@ -729,4 +729,40 @@ module.exports = Facility => {
...
+  }
+
+
+  Facility.remoteMethod("findBySystem", {
+    description: 'retrieve by system',
+    http: { path: '/findBySystem', verb: "get" },
+    accepts: [{ arg: 'system', type: 'string' }, { arg: 'code', type: 'string' }],
```
Comment (bwighane) [inline common/models/facility.js]: Make system and code as required. I tried making a wrong query and the app crashed. Better force the consumer to make sure they pass through all required arguments. simply add required: true for each accepts.

**Review by danielmwakanema — APPROVED** [review]
Looks good to me at the moment. Only concern would be the performance of the string ops though at this stage I haven't tried it.

**Review by bwighane — CHANGES_REQUESTED** [review]

File: common/models/facility.js (hunk 1 above)
Comment (bwighane) [reply common/models/facility.js]: *string format

Comment (Loui4) [reply common/models/facility.js]: will work on it

Comment (Loui4) [reply common/models/facility.js]: will resolve that

File: common/models/facility.js (hunk 2 above)
Comment (Loui4) [reply common/models/facility.js]: will do that as well

**Comment by Loui4** [comment]
check out the changes

**Review by bwighane — APPROVED** [review]
This looks great, nice work.

**Pull request merged**